include/CL/sycl/buffer/detail/buffer.hpp

```diff
@@ -133,10 +198,24 @@ class buffer : public detail::buffer_base,
     // The size of a multi_array is set at creation time
     allocation { boost::extents[std::distance(start_iterator, end_iterator)] },
     access { allocation }
+    // If iterators are const ones, then we do not write back
```

Comment:
It is more complex than this I guess http://en.cppreference.com/w/cpp/iterator/iterator_tags